src/libsync/syncoptions.cpp

```diff
@@ -69,10 +82,35 @@ void SyncOptions::fillFromEnvironmentVariables()
         _parallelNetworkJobs = maxParallel;
 }
 
-void SyncOptions::verifyChunkSizes()
+void SyncOptions::fillFromAccount(const AccountPtr account)
+{
+    if (account) {
+        if (account->isHttp2Supported() && _parallelNetworkJobs == defaultParallelNetworkJobs) {
+            _parallelNetworkJobs = defaultParallelNetworkJobsH2;
+        }
+
+        if (const auto size = account->getMaxRequestSize(); size > 0) {
+            setMaxChunkSize(size);
+        }
+
+        if (account->capabilities().chunkingNg()) {
+            // read last successfully used chunk size and use it as initial value
+            if (const auto size = account->getLastChunkSize(); size > 0) {
+                setInitialChunkSize(size);
+            }
+        } else {
+            // disable dynamic chunk sizing as it is not supported for this account
+            _targetChunkUploadDuration = std::chrono::milliseconds(0);
+        }
+    }
```

Comment:
nitpick: I would do an early return here instead and remove one indent level from the rest of the method